src/Plugin/RulesAction/SystemMailToUsersOfRole.php

```diff
@@ -0,0 +1,173 @@
+ *       label = @Translation("Roles"),
+ *       description = @Translation("The roles to which to send the e-mail."),
+ *       multiple = TRUE
+ *     ),
+ *     "subject" = @ContextDefinition("string",
+ *       label = @Translation("Subject"),
+ *       description = @Translation("The subject of the e-mail."),
+ *     ),
+ *     "body" = @ContextDefinition("string",
+ *       label = @Translation("Body"),
+ *       description = @Translation("The body of the e-mail."),
+ *     ),
+ *     "from" = @ContextDefinition("email",
+ *       label = @Translation("From"),
+ *       description = @Translation("The from e-mail address."),
+ *       required = FALSE
+ *     )
+ *   }
+ * )
+ *
+ * @todo: Add access callback information from Drupal 7.
+ */
+class SystemMailToUsersOfRole extends RulesActionBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The logger channel the action will write log messages to.
+   *
+   * @var \Psr\Log\LoggerInterface
+   */
+  protected $logger;
+
+  /**
+   * Mail manager.
+   *
+   * @var \Drupal\Core\Mail\MailManagerInterface
+   */
+  protected $mailManager;
+
+  /**
+   * The site configuration.
+   *
+   * @var \Drupal\Core\Config\ImmutableConfig
+   */
+  protected $config;
+
+  /**
+   * The user storage service.
+   *
+   * @var \Drupal\user\UserStorage
+   */
+  protected $userStorage;
+
+  /**
+   * Constructs a SendMailToUsersOfRole object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Psr\Log\LoggerInterface $logger
+   *   The storage service.
+   * @param \Drupal\Core\Mail\MailManagerInterface $mail_manager
+   *   The mail manager service.
+   * @param \Drupal\Core\Config\ImmutableConfig $site_config
+   *   The site configuration.
+   * @param \Drupal\user\UserStorage $user_storage
+   *   The user storage service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, LoggerInterface $logger, MailManagerInterface $mail_manager, ImmutableConfig $site_config, UserStorageInterface $user_storage) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->logger = $logger;
+    $this->mailManager = $mail_manager;
+    $this->config = $site_config;
+    $this->userStorage = $user_storage;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('logger.factory')->get('rules'),
+      $container->get('plugin.manager.mail'),
+      $container->get('config.factory')->get('system.site'),
+      $container->get('entity.manager')->getStorage('user')
```

Comment:
we shouldn't use the entity.manager because it is deprecated. Use entity type manager instead.